compiler/src/dmd/funcsem.d

```diff
@@ -3100,6 +3100,247 @@ extern (D) void checkMain(FuncDeclaration fd)
         .error(fd.loc, "%s `%s` must return `int`, `void` or `noreturn`, not `%s`", fd.kind, fd.toPrettyChars, tf.nextOf().toChars());
 }
 
+enum LevelError = -2;
+
+/*****************************************
+ * Determine lexical level difference from `fd1` to nested function `fd2`.
+ * Params:
+ *      fd2 = target of call
+ *      intypeof = !=0 if inside typeof
+ * Returns:
+ *      0       same level
+ *      >0      decrease nesting by number
+ *      -1      increase nesting by 1 (`fd2` is nested within `fd1`)
+ *      LevelError  error, `this` cannot call `fd2`
+ */
+extern (D) final int getLevel(FuncDeclaration fd1, FuncDeclaration fd2, int intypeof)
```

Comment:
remove the `final`, and update the `Params:` section of the documentation above to include `fd1 = function` (or a better description if you can come up with one).